python/lsst/obs/base/ingest.py

```diff
@@ -237,50 +272,62 @@ def extractMetadata(self, filename: str) -> RawFileData:
         # Instead return a RawFileData with no datasets and allow
         # the caller to report the failure.
 
+        # The options for extracting metadata are:
+        # - This is a JSON index file with metadata so use it directly
+        #   (assumes the file is then not listed explicitly)
+        # - Read metadata from a _index.json file in the same directory as the
+        #   non-JSON file. Should be cached. This is tricky when process pools
+        #   are used since the same index file could be found by
+        #   multiple processes.
+        # - Read metadata from a sidecar .json file (a file of the same name
+        #   as the data file but with .json extension)
+        # - Read metadata from the file itself
+
+        used_sidecar = False
         try:
-            # Manually merge the primary and "first data" headers here because
-            # we do not know in general if an input file has set INHERIT=T.
-            phdu = readMetadata(filename, 0)
-            header = merge_headers([phdu, readMetadata(filename)], mode="overwrite")
+            root, ext = os.path.splitext(filename)
+            sidecar_file = root + ".json"
+            if os.path.exists(sidecar_file):
+                header = read_sidecar(sidecar_file)
+                used_sidecar = True
+            else:
+                # Read the metadata from the data file itself
+                # Manually merge the primary and "first data" headers here
+                # because we do not know in general if an input file has
+                # set INHERIT=T.
+                phdu = readMetadata(filename, 0)
+                header = merge_headers([phdu, readMetadata(filename)], mode="overwrite")
             datasets = [self._calculate_dataset_info(header, filename)]
         except Exception as e:
```

Comment:
We want every possible problem to be caught since these failures are associated with specific file issues. If we have a coding bug that leads to every file being rejected then the ingest code would make that clear at the end. The ingest failures are reported at the end of the ingest and a callback is also used to allow the caller to inspect the exception. I can changed the debug to info but I worry that this will overwhelm people.